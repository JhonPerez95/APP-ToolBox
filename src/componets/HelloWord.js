import React from 'react';
import useForm from '../hooks/useForm';

const HelloWord = () => {
  const [formValue, handleInputChange, resetInput] = useForm({
    text: '',
  });
  const { text } = formValue;
  const handleSubmit = (e) => {
    e.preventDefault();
    console.log(formValue);
  };
  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Ingreso</h3>
          <form onSubmit={handleSubmit}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Send Text"
                name="text"
                value={text}
                onChange={handleInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="submit"
                className="btnSubmit btn btn-primary"
                value="Text"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default HelloWord;
