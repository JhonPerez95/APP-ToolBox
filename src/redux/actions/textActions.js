import { fetchApi } from '../../helpers/fetchApi';

const { types } = require('../types/types');

const sendText = (text) => {
  return {
    type: types.textSend,
    payload: text,
  };
};

export const StartSendText = (text) => {
  return async (dispatch) => {
    try {
      const resp = await fetchApi('hellow', text, 'POST');
      const body = await resp.json();
      console.log(body);
      const { data } = body;
      dispatch(sendText(data.text));
    } catch (error) {
      console.log(error);
    }
  };
};
