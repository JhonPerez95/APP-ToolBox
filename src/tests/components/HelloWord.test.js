import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { StartSendText } from '../../redux/actions/textActions';
import HelloWord from '../../componets/HelloWord';

// Config Store Redux
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = { text: { textField: 'test the input' } };
let store = mockStore(initState);
store.dispatch = jest.fn();

// Simulation action
jest.mock('../../redux/actions/textActions', () => ({
  StartSendText: jest.fn(),
}));

describe('Test the component HelloWord', () => {
  // Refresh store and mock
  beforeEach(() => {
    store = mockStore(initState);
    jest.clearAllMocks();
  });
  const wrapper = mount(
    <Provider store={store}>
      <HelloWord />
    </Provider>
  );

  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should  call action StartSendText', () => {
    wrapper.find('form').prop('onSubmit')({
      preventDefault() {},
    });
    expect(StartSendText).toHaveBeenCalled();
    expect(StartSendText).toHaveBeenCalledWith({ text: '' });
  });

  test('should show  Alert', () => {
    const { text } = initState;
    const divAlert = wrapper.find('.alert');
    // console.log(divAlert.text());
    expect(divAlert.exists()).toBe(true);
    expect(divAlert.text()).toBe(text.textField);
  });
});
