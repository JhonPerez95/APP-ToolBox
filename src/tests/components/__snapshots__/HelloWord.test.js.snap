// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component HelloWord should render component 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Prueba Tecnica:
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="text"
            onChange={[Function]}
            placeholder="Send Text"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit btn btn-primary"
            type="submit"
            value="Send"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Response API:
      </h3>
      <div
        className="alert alert-success"
        role="alert"
      >
        test the input
      </div>
    </div>
  </div>
</div>
`;
